types/serve-handler/index.d.ts

```diff
@@ -0,0 +1,217 @@
+// Type definitions for serve-handler 6.1
+// Project: https://github.com/zeit/serve-handler#readme
+// Definitions by: simonhaenisch <https://github.com/simonhaenisch>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { lstat, realpath, createReadStream, readdir } from 'fs';
+import { IncomingMessage, ServerResponse } from 'http';
+
+export default serveHandler;
```

Comment:
This needs `export =` instead of `export default`. Refer to https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

Reply:
I read that and had that first, however it seemed like it didn't apply in this case. The package has

```js
module.exports = async (/* ... */) => { /* ... */ }
```

So would `import * as serveHandler from 'serve-handler'` really be the correct way of importing this module (without synthetic default imports)?

Reply:
If I do `export =`, and then in my test file (`serve-handler/serve-handler-tests.ts`) I do

```ts
import serveHandler from 'serve-handler'
```

I get

> This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

If I do 

```ts
import * as serveHandler from 'serve-handler'
```

I get

> This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.

From how I understood it, `esModuleInterop` is supposed to be on by default, is it not?

Reply:
`module.exports =` gets converted to `export = ` syntax in typescript.  

Running code
```ts
const x = require("serve-handler");
console.log(x.default);
```
```
> node a.js
undefined
```

Reply:
Oh my bad, I see now that I'm supposed to import it as

```ts
import serveHandler = require('serve-handler');
```

And it works now to set `allowSyntheticDefaultImports` in `types/serve-handler/tsconfig.json`, and use `import serveHandler from 'serve-handler'`... for some reason that wasn't working before, maybe my VS Code was just hanging 🤷🏻‍♂️.